### PR TITLE
issue/3841 - Migrate logs to custom table

### PR DESCRIPTION
**Pull request by sunnyratilal**
**WIP - Do not merge**

Issue: #3841

### CODE REVIEW

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -1349,4 +1364,232 @@ function edd_remove_legacy_discounts() {
...
+					'price_id'     => isset( $post_meta['_edd_log_price_id'] ) ? $post_meta['_edd_log_price_id'] : 0,
+					'user_id'      => isset( $post_meta['_edd_log_user_id'] ) ? $post_meta['_edd_log_user_id'] : 0,
+					'ip'           => $post_meta['_edd_log_ip'],
+					'date_created' => $old_log->post_date,
+				);
+
+				$new_log_id = EDD()->file_download_logs->insert( $log_data );
+
+				if ( ! empty( $new_log_id ) ) {
+					add_post_meta( $old_log->ID, '_edd_log_migrated_id', $new_log_id );
```
Comment (pippinsplugins) [inline includes/admin/upgrades/upgrade-functions.php]: Why are we adding new meta data to logs we're getting out of post?

File: includes/admin/upgrades/upgrade-functions.php [hunk 2]
```diff
@@ -1349,4 +1364,232 @@ function edd_remove_legacy_discounts() {
...
+	$log_ids = $wpdb->get_results( "SELECT ID FROM $wpdb->posts WHERE post_type = 'edd_log'" );
+	$log_ids = wp_list_pluck( $log_ids, 'ID' );
+	$log_ids = implode( ', ', $log_ids );
+
+	edd_debug_log( 'Beginning removal of legacy logs' );
+
+	if ( ! empty( $log_ids ) ) {
+		$delete_posts_query = "DELETE FROM $wpdb->posts WHERE ID IN ({$log_ids})";
+		$wpdb->query( $delete_posts_query );
+		$delete_postmeta_query = "DELETE FROM $wpdb->postmeta WHERE post_id IN ({$log_ids})";
```
Comment (pippinsplugins) [inline includes/admin/upgrades/upgrade-functions.php]: Do we know for certain this will work on sites that have 2,000,000+ log entries?

Comment (sunnyratilal) [reply includes/admin/upgrades/upgrade-functions.php]: I think this query will be better if I change it to `DELETE FROM $wpdb->posts WHERE post_type = 'edd_log'`

Comment (pippinsplugins) [reply includes/admin/upgrades/upgrade-functions.php]: I agree

**Pull request merged**